src/render/painter.js

```diff
@@ -555,6 +555,7 @@ class Painter {
         // With terrain on, renders the depth buffer into a texture.
         // This texture is used for occlusion testing (labels)
         if (this.terrain && (this.style.hasSymbolLayers() || this.style.hasCircleLayers())) {
+            // $FlowFixMe[incompatible-use] - Flow can't infer that terrain is not null
```

Comment:
Nit: in cases like this, usually it helps to save `this.terrain` in a variable — Flow can't know that the value of a property didn't change after the `if` check.